File: lang/python/test/muse/algorithms/sorting/MergeSortTest.py
from muse.algorithms.sorting import MergeSort
from muse.util import SequenceBuilder
from muse.util import Sequences
from muse.util import TestRunner

def testMergeSort() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    MergeSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum:
        return False

    return Sequences.isSorted(arr)

def main() -> None:
    TestRunner.parseTest(testMergeSort())

if __name__ == "__main__":
    main()

File: ruby/test/muse/algorithms/sorting/MergeSortTest.rb
require "muse/algorithms/sorting/MergeSort"
require "muse/util/SequenceBuilder"
require "muse/util/Sequences"
require "muse/util/TestRunner"

def testMergeSort()
    size = 32768

    arr = Array.new(size)
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    MergeSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum
        return false
    end

    Sequences.isSorted(arr)
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.parseTest(testMergeSort())
end
